### DISCUSSION TITLE
how do I create a dev.spacy file?

### DISCUSSION THREAD

**Discussion by DarkSoliditi** [explosion/spaCy#9000] [Help: Coding & Implementations] [answered]
Hey I'm very new to these concepts & I've looked for this question in particular and can't find an answer, If I have duplicated a existing thread I apologize.

I think I understand how to make the train.spacy file to be used in the training CLI and there is alot of documentation on making it but I can't find any documentation on making the dev.spacy.

What kind of information needs to be in the dev.spacy file & what is the best way to make it?

**Comment by polm** [comment, marked as answer]
A `dev.spacy` file is the same as a `train.spacy` file, it just contains a different set of examples. 

When training a model, you periodically want to evaluate it to see how well it's doing. If you evaluate it on the data you've been training it on, you can't tell if it's getting a good score because it's actually learned how to work correctly or if it's just memorized the examples you've shown it. To keep the model honest and tell when it's memorizing things, it's traditional to use validation data or a "dev set" of examples for periodic evaluation during training. 

Normally you would create a dev set by taking your initial data and randomly splitting it into train and dev (or train and dev and test) sets. 90% train / 10% dev (or 80/10/10) is pretty common.

After you have split the data, you make the `dev.spacy` file exactly the same way you'd make the `train.spacy` file.

**Reply by DarkSoliditi** [reply]
Awesome, that clears it up. Thanks so much!
